Library/Homebrew/dev-cmd/release.rb

```diff
@@ -0,0 +1,119 @@
+      conflicts "--major", "--minor"
+
+      named_args :none
+    end
+  end
+
+  def release
+    args = release_args.parse
+
+    safe_system "git", "-C", HOMEBREW_REPOSITORY, "fetch", "origin"
+    local_head = Utils.popen_read("git", "-C", HOMEBREW_REPOSITORY, "rev-parse", "HEAD").chomp
+    upstream_head = Utils.popen_read("git", "-C", HOMEBREW_REPOSITORY, "rev-parse", "origin/HEAD").chomp
+    if local_head != upstream_head
+      message = "The current HEAD does not match origin/HEAD! Pass --force-mismatched-head to override."
+      odie message unless args.force_mismatched_head?
+
+      opoo message
+    end
+
+    current_version = Utils.popen_read(
+      "git", "-C", HOMEBREW_REPOSITORY, "tag", "--list", "--sort=-version:refname", "*.*.*"
+    ).lines.first.chomp.presence
+    odie "No existing tags found!" if current_version.blank?
+    current_version = Version.new current_version
+
+    if args.major? || args.minor?
+      one_month_ago = Date.today << 1
+      current_version_date = Date.parse Utils.popen_read(
+        "git", "-C", HOMEBREW_REPOSITORY, "log", "-1", "--format=%aI", "#{current_version.major_minor}.0"
+      )
+      message = "The latest major/minor release was less than one month ago. Pass --force-major-minor to override."
+      if current_version_date > one_month_ago
+        odie message unless args.force_major_minor?
+
+        opoo message
+      end
+    end
+
+    new_version = if args.major?
+      Version.new [current_version.major.to_i + 1, 0, 0].join(".")
+    elsif args.minor?
+      Version.new [current_version.major, current_version.minor.to_i + 1, 0].join(".")
+    else
+      Version.new [current_version.major, current_version.minor, current_version.patch.to_i + 1].join(".")
+    end.to_s
+
+    ohai "Tagging version #{new_version}"
+    safe_system "git", "-C", HOMEBREW_REPOSITORY, "tag", "--sign", new_version, "-m", new_version
+
+    return if args.no_push?
+
+    ohai "Pushing Tags"
+    # FIXME: Change from `Rylan12` to `origin` before creating a PR
+    safe_system "git", "-C", HOMEBREW_REPOSITORY, "push", "Rylan12", new_version
+
+    ohai "Creating Release"
+    release_notes = if args.major? || args.minor?
+      ["Release notes for this release can be found on the [Homebrew blog](https://brew.sh/blog/#{new_version})."]
+    else
+      []
+    end
+    release_notes += Utils.popen_read(
+      "git", "-C", HOMEBREW_REPOSITORY, "log", "--pretty=format:'%s >> - %b%n'", "#{current_version}..#{new_version}"
+    ).lines.grep(/Merge pull request/).map! do |s|
+      pr = s.gsub(%r{.*Merge pull request #(\d+) from ([^/]+)/[^>]*(>>)*},
+                  "https://github.com/Homebrew/brew/pull/\\1 (@\\2)")
+      /(.*\d)+ \(@(.+)\) - (.*)/ =~ pr
+      "- [#{Regexp.last_match(3)}](#{Regexp.last_match(1)}) (@#{Regexp.last_match(2)})"
+    end
+
+    begin
+      release = GitHub.create_release tag_name: new_version, name: new_version, body: release_notes.join("\n"), args: args
```

Comment:
Could consider creating just a draft release, not pushing the tag and allowing the user to use the GitHub UI to publish the release (which will create the tag).